src/components/select/select.tsx

```diff
@@ -139,23 +145,18 @@ export class Select implements FormComponent {
   @Listen("gxSelectDidLoad")
   onSelectOptionDidLoad(ev: HTMLSelectOptionElementEvent) {
     const option = ev.target;
-    this.updateOptions(this.getChildOptions());
-
-    if (this.value !== undefined && option.value === this.value) {
-      // this select has a value and this
-      // option equals the correct select value
-      // so let's check this option
-      option.selected = true;
-    } else if (this.value === undefined && option.selected) {
-      // this select does not have a value
-      // but this option is checked, so let's set the
-      // select's value from the checked option
-      this.value = option.value;
-    } else if (option.selected) {
-      // if it doesn't match one of the above cases, but the
-      // option is still checked, then we need to uncheck it
-      option.selected = false;
+    if (this.value) {
+      if (this.value === option.value) {
+        option.selected = true;
+      } else {
+        option.selected = false;
+      }
+    } else {
+      if (option.selected) {
+        this.value = option.value;
+      }
```

Comment:
Why did we remove all the comments? I think they were relevant.

Reply:
I thought it was pretty intuitive...
do you say to add them again?

Reply:
@ncardeli comments are back!
Check them in f6a05fe